app/services/breadcrumbs_options.rb

```diff
@@ -20,11 +20,12 @@ def bar_props
   private
 
   delegate :logger, :to => :Rails
-  attr_reader :controller_name, :action_name, :page_header, :options
+  delegate :resource_name, :resource_class, :controller_path, :to => :controller
+  attr_reader :controller, :action_name, :page_header, :options
   def index_item
     {
-      caption: _(controller_name.humanize),
-      url: try("#{controller_name}_path")
+      caption: _(resource_name.pluralize.humanize),
+      url: try("#{resource_name.pluralize}_path")
```

Comment:
Does it make sense to differentiate `resource_path` (a new method for the controllers) and `resource_name`?
If the path is namespaced, we need to use `foreman_whatever_models_path` here. But `find_resource` will then set an instance variable `@foreman_whatever_model`. But I guess devs want to just use the instance variable `@model` (without the namespace).

Reply:
Probably makes sense, is `resource_path` always available?

Reply:
Actually I just invented that name, but we have some interesting code here: https://github.com/theforeman/foreman/blob/4e6d22620ed3bbd9e1caa5652f2e0df5ebe21d61/app/helpers/filters_helper.rb#L23-L36

Reply:
great, I took that method and put it into ApplicationController as a helper_method and used at `index_item` but not here, because it wouldn't work with a plugin that does not have namespaced controller class but has namespaced API, such as foreman_openscap

I now added a test for whetner `resource_name + '_path'` is defined and use it in such case (e.g. `policy_path :id => ':id' # => /compliance/policy/:id`) and if not, we use `"/#{controller_path}/:id"` which works for plugin such as virt_who_configure and non-namespaced plugins.

mind to take another look @timogoebel